src/main/kotlin/org/rust/ide/intentions/NestUseStatementsIntention.kt

```diff
@@ -0,0 +1,207 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.RsUseGroup
+import org.rust.lang.core.psi.RsUseItem
+import org.rust.lang.core.psi.RsUseSpeck
+import org.rust.lang.core.psi.ext.*
+
+class NestUseStatementsIntention : RsElementBaseIntentionAction<NestUseStatementsIntention.Context>() {
+    override fun getText() = "Nest use statements"
+    override fun getFamilyName() = text
+
+    interface Context {
+        val useSpecks: List<RsUseSpeck>
+        fun root(): PsiElement
+        fun firstPreviousElement(): PsiElement
```

Comment:
Consider "firstOldElement()"/"getOldElements()".
Also consider convert some these functions to properties. In some cases it will reduce code duplication (just like with `useSpecks`)